src/exe.ml

```diff
@@ -124,6 +134,12 @@ let link_exe
       ?(js_of_ocaml=Dune_file.Js_of_ocaml.default)
       cctx
   =
+  let linkage, js =
+    if linkage.ext = ".bc.js" then
+      {linkage with ext = ".bc"}, true
+    else
+      linkage, false
+  in
```

Comment:
What happens if we use `(modes byte js)`? It seems to me that the rule for the `.bc` is going to be produced twice.

Reply:
You are right; I will fix this.

Reply:
I pushed a commit that should fix this issue; it also avoids generating `.cmo.js` rules when in explicit js mode if js compilation has not been requested.

Reply:
Ok, not generating the .cmo.js rules seems good.